modules/library/coverage/src/main/java/org/geotools/coverage/processing/OperationJAI.java

```diff
@@ -1110,28 +1108,24 @@ protected static void handleROINoDataInternal(
         // Getting the internal ROI property
         ROI innerROI = CoverageUtilities.getROIProperty(sourceCoverage);
         // Checking if it canbe set as parameter
-        if (JAIExt.isJAIExtOperation(operationName) && roiIndex >= 0) {
-            // Definition of the nwe ROI
-            ROI roiParam = (ROI) parameters.getObjectParameter(roiIndex);
-            ROI newROI = null;
-            if (innerROI == null) {
-                newROI = roiParam;
-            } else {
-                newROI = roiParam != null ? innerROI.intersect(roiParam) : innerROI;
-            }
-            // Setting of the new ROI
-            parameters.set(newROI, roiIndex);
+        // Definition of the nwe ROI
+        ROI roiParam = (ROI) parameters.getObjectParameter(roiIndex);
+        ROI newROI = null;
+        if (innerROI == null) {
+            newROI = roiParam;
+        } else {
+            newROI = roiParam != null ? innerROI.intersect(roiParam) : innerROI;
         }
+        // Setting of the new ROI
+        parameters.set(newROI, roiIndex);
 
         // Getting NoData propery
         NoDataContainer nodataProp = CoverageUtilities.getNoDataProperty(sourceCoverage);
         Range innerNodata = nodataProp != null ? nodataProp.getAsRange() : null;
         // Setting the NoData Range parameter if not present
-        if (JAIExt.isJAIExtOperation(operationName) && noDataIndex >= 0) {
```

Comment:
Should we preserve the noDataIndex >=0 check ?

Reply:
Yep